src/FTPObject.jl

```diff
@@ -132,168 +132,156 @@ end
 """
     upload(
         ftp::FTP,
-        local_name::AbstractString;
+        local_path_io::IO,
+        remote_path::AbstractString;
+        ftp_options=ftp.ctxt,
         mode::FTP_MODE=binary_mode,
-    )
+        verbose=nothing
+    ) -> Bool
 
-Upload the file "local_name" to the FTP server and save as "local_name".
-"""
-function upload(
-    ftp::FTP,
-    local_name::AbstractString;
-    mode::FTP_MODE=binary_mode,
-    verbose=nothing,
-)
-    _dep_verbose_kw(verbose, typeof(ftp), :upload)
-    return upload(ftp, local_name, local_name; mode=mode, verbose=verbose)
-end
+Upload IO object "local_path_io" to the FTP server and save as "remote_path".
 
-"""
-    upload(
-        ftp::FTP,
-        local_name::AbstractString,
-        remote_name::AbstractString;
-        mode::FTP_MODE=binary_mode,
-    )
+# Arguments
+`ftp::FTP`: The FTP to deliver to. See FTPClient.FTP for details.
+`local_path_io::IO`: The IO object that we want to deliver.
+`remote_path::AbstractString`: The path that we want to deliver to.
+
+# Keywords
+`ftp_options=ftp.ctxt`: FTP Options
+`mode::FTP_MODE=binary_mode`: Set the ftp mode.
+`verbose=nothing`: Set the verbosity
 
-Upload the file "local_name" to the FTP server and save as "remote_name".
+# Returns
+`Bool`: Returns a boolean with true if the file was successfully transfered, else false.
 """
 function upload(
     ftp::FTP,
-    local_name::AbstractString,
-    remote_name::AbstractString;
+    local_path_io::IO,
+    remote_path::AbstractString;
+    ftp_options=ftp.ctxt,
     mode::FTP_MODE=binary_mode,
     verbose=nothing,
-)
+)::Bool
```

Comment:
Maybe we should just return the response code?

Reply:
would it make sense to return a bool of success/fail and also the response data or code?

Reply:
If you have the response code you know whether the request succeeded or failed, so having the boolean in addition to the response code seems redundant to me.

Reply:
I suppose I'd be fine with that, but I feel like just returning the response or the code is sufficient. I don't really see an advantage to returning a bool when folks could just look up the return code.

Reply:
ok that makes sense.